docs/root/configuration/listeners/udp_filters/dns_filter.rst

```diff
@@ -54,8 +63,71 @@ Example Configuration
               endpoint:
                 address_list:
                   address:
-                    - 10.0.3.1
+                  - 10.0.3.1
 
 
 In this example, Envoy is configured to respond to client queries for three domains. For any
-other query, it will forward upstream to external resolvers.
+other query, it will forward upstream to external resolvers. The filter will return an address
+matching the input query type. If the query is for type A records and no A records are configured,
+Envoy will return no addresses and set the response code appropriately. Conversely, if there are
+matching records for the query type, each configured address is returned. This is also true for
+AAAA records. Only A and AAAA records are supported. If the filter parses other queries for other
```

Comment:
Do we have validation for a number of records that are > the size of a datagram?

Reply:
Yes. I have tests where this is checked. They will be in a future PR.

Reply:
There is a 1024 udp datagram size limit elsewhere in the code (don't remember off the top of my head), but the filter does enforce the 512 byte limit when serializing answers.   I do have a note in the code that refers to the existing 1024 byte limit and we will have to work with/around that if we end up supporting extended DNS options later.

Reply:
OK sounds good, thanks.